Source/WebCore/platform/graphics/skia/ColorSpaceSkia.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2024 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "ColorSpaceSkia.h"
+
+#include "NotImplemented.h"
+#include "PlatformColorSpace.h"
+
+#if USE(SKIA)
+
+namespace WebCore {
+
+sk_sp<SkColorSpace> sRGBColorSpaceRef()
+{
+    return SkColorSpace::MakeSRGB();
+}
+
+sk_sp<SkColorSpace> linearSRGBColorSpaceRef()
+{
+    return SkColorSpace::MakeSRGBLinear();
+}
+
+#if ENABLE(DESTINATION_COLOR_SPACE_DISPLAY_P3)
+sk_sp<SkColorSpace> displayP3ColorSpaceRef()
+{
+    return SkColorSpace::MakeRGB(SkNamedTransferFn::kSRGB, SkNamedGamut::kDisplayP3);
+}
+#endif
+
+int PlatformColorSpace::serialize() const
+{
+    notImplemented();
+    return 0;
+}
+
+void PlatformColorSpace::deserialize(int)
+{
+    notImplemented();
+}
```

Comment:
I guess we can add these when needed, no? because this looks specific to skia.

Reply:
I tried pretty hard to not include that but I couldn't find a way to avoid serialization in `Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in`. I have an impression that this is mandatory - e.g. in cairo case we're using a value of `enum class Name : uint8_t` and in case of CG we're using RefPtr's MO.

Reply:
Ok